File: src/dependent/fulfillment/ShippingService.java
package org.example.dependent.fulfillment;

public class ShippingService {
    public String createShipment(String orderId, Address address) {
        if (orderId == null || orderId.isEmpty()) {
            throw new IllegalArgumentException("orderId must be provided");
        }
        if (address == null) {
            throw new IllegalArgumentException("address must be provided");
        }
        return "trk-" + Math.abs((orderId + address.toString()).hashCode());
    }
}

File: src/dependent/fulfillment/shipping_service.py
from .address import Address


class ShippingService:
    def create_shipment(self, order_id: str, address: Address) -> str:
        if order_id is None or order_id == "":
            raise ValueError("orderId must be provided")
        if address is None:
            raise ValueError("address must be provided")
        return "trk-" + str(abs(hash(order_id + str(address))))
